runtime/v2/shim/publisher.go

```diff
@@ -43,12 +44,22 @@ type item struct {
 
 // NewPublisher creates a new remote events publisher
 func NewPublisher(address string) (*RemoteEventsPublisher, error) {
+	return newPublisher(context.Background(), address)
+}
+
+// NewPublisherWithContext creates a new remote events publisher with context
+func NewPublisherWithContext(ctx context.Context, address string) (*RemoteEventsPublisher, error) {
```

Comment:
It would be better to add the context parameter directly to the NewPublisher.
I'm not sure if it's possible to modify the public functions directly in 2.0